app/models/concerns/assign_to_current_semester.rb

```diff
@@ -6,6 +6,7 @@ module AssignToCurrentSemester
   end
 
   def assign_to_current_semester
-    self.semester = Semester.current_semester unless semester.present?
+    # TODO: Settings are not initilized yet here
+    # self.semester = Semester.find(Settings.new.current_semester_id)
```

Comment:
hm. it seems like there should always be at least 1 semester created before any apps are created.

making sure that there is at least one semester in the app at all times seems to fall back into the same issue with Settings where we need to create a semester in seeds or we create a dummy semester in ApplicationController maybe